classifieds/classifieds-library/src/main/java/org/silverpeas/components/classifieds/service/DefaultClassifiedService.java

```diff
@@ -610,7 +599,21 @@ public void deleteAllSubscribes(String instanceId) {
   }
 
   @Override
-  public Collection<ClassifiedDetail> getAllValidClassifieds(String instanceId,
+  public List<ClassifiedDetail> getAllValidClassifieds(String instanceId) {
+    Connection con = openConnection();
+    try {
+      return ClassifiedsDAO
+          .getClassifiedsWithStatus(con, instanceId, ClassifiedDetail.VALID, -1, -1);
+    } catch (Exception e) {
+      throw new ClassifiedsRuntimeException("DefaultClassifiedService.getAllValidClassifieds()",
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "DefaultClassifiedService.getAllValidClassifieds()" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)